### PR TITLE
Introduce new parameter max_survivable_failed_members.

### PR DESCRIPTION
**Pull request by mneedham**
This parameter is useful for some very large deployments.

### CODE REVIEW

File: enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/atomicbroadcast/multipaxos/ProposerState.java [hunk 1]
```diff
@@ -347,6 +347,8 @@ else if ( instance.isState( PaxosInstance.State.closed )
                                                         org.neo4j.cluster.protocol.atomicbroadcast.multipaxos.InstanceId.INSTANCE ) );
                                             }
                                         }
+                                        System.out.printf( "%s sent learn messages to %d members.%n",
```
Comment (digitalstain) [inline enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/atomicbroadcast/multipaxos/ProposerState.java]: We probably don't want that here.

File: enterprise/cluster/src/main/java/org/neo4j/cluster/ClusterSettings.java [hunk 2]
```diff
@@ -142,5 +142,12 @@ public String toString()
     @Description( "Timeout for waiting for other members to finish a role election. Defaults to ha.paxos_timeout." )
     public static final Setting<Long> election_timeout = setting( "ha.election_timeout", DURATION, paxos_timeout );
 
+    @Description( "Maximum number of servers which can fail without making the whole cluster unavailable. " +
+            "In very large clusters, the probability of half the cluster failing is low, but protecting against " +
+            "any arbitrary half failing is expensive. Therefore setting this parameter to less than half the cluster " +
+            "size is useful to speed up consensus, without compromising availability. ")
+    public static final Setting<Integer> max_survivable_failed_members =
+            setting( "ha.max_survivable_failed_members", INTEGER, "1" );
```
Comment (lutovich) [inline enterprise/cluster/src/main/java/org/neo4j/cluster/ClusterSettings.java]: Maybe limit this value with `Settings#min()` and `Settings#max()`?

File: enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/atomicbroadcast/multipaxos/ProposerState.java (hunk 1 above)
Comment (mneedham) [inline enterprise/cluster/src/main/java/org/neo4j/cluster/protocol/atomicbroadcast/multipaxos/ProposerState.java]: D'oh. How did that sneak in?! WIll fix

File: enterprise/cluster/src/main/java/org/neo4j/cluster/ClusterSettings.java (hunk 2 above)
Comment (mneedham) [inline enterprise/cluster/src/main/java/org/neo4j/cluster/ClusterSettings.java]: Oh good idea, didn't know about that. Will do that

**Comment by spacecowboy** [comment]
@mneedham this PR should undergo HA-robustness before going in

**Comment by spacecowboy** [comment]
ha-robustness was success: https://build.neohq.net/viewLog.html?buildId=688799&tab=buildResultsDiv&buildTypeId=JonasHaRequests_CustomHaRobustness

**Pull request merged**